lib/Cucumber.j

```diff
@@ -881,103 +894,132 @@ function dumpGuiObject(obj)
     else
         locationWindowPoint = CGPointMake(CGRectGetMidX([obj frame]), CGRectGetMidY([obj frame]));
 
-    CPLog.debug("Cucapp is about to simulate a left click on the view : " + obj);
+    var deltaX = params.shift(),
+        deltaY = params.shift(),
+        flags = params.shift(),
+        modifierFlags = 0;
+
+    for (var i = 0; i < [flags count]; i++)
+    {
+        var flag = flags[i];
+        modifierFlags |= parseInt(flag);
+    }
 
-    [self _perfomMouseEventOnPoint:locationWindowPoint toPoint:nil window:[obj window] eventType:CPLeftMouseDown numberOfClick:1 modifierFlags:params[0]];
+    CPLog.debug("Cucapp is about to simulate a scroll wheel on the view : " + obj + " with the deltas : " + deltaX + "," + deltaY + " and modifiers flags " + modifierFlags);
+
+    var event = [self _createMouseEventWithType:@"DOMMouseScroll" target:[self _mainDOMDocument] button:button clientX:locationWindowPoint.x clientY:locationWindowPoint.y ctrlKey:modifierFlags & CPControlKeyMask shiftKey:modifierFlags & CPShiftKeyMask altKey:modifierFlags & CPAlternateKeyMask metaKey:modifierFlags & CPCommandKeyMask];
+    event["deltaX"] = deltaX;
+    event["deltaY"] = deltaY;
+
+    [self _mainDOMWindow].dispatchEvent(event);
+    [[CPRunLoop currentRunLoop] limitDateForMode:CPDefaultRunLoopMode];
 
     return '{"result" : "OK"}';
 }
 
-- (void)simulateRightClickOnPoint:(CPArray)params
+- (void)simulateMouseDownOnPoint:(CPArray)params
 {
-    var point = CGPointMake(params.shift(), params.shift()),
+    var locationWindowPoint = CGPointMake(params.shift(), params.shift()),
+        flags = params[0],
+        mouseType = params[1],
         window = [CPApp keyWindow];
 
-    CPLog.debug("Cucapp is about to simulate a right click on the point : " + point);
+    if (mouseType != CPLeftMouseDown && mouseType != CPRightMouseDown)
+        return '{"result" : "INVALID MOUSE TYPE"}';
 
-    [self _perfomMouseEventOnPoint:point toPoint:nil window:window eventType:CPRightMouseDown numberOfClick:1 modifierFlags:params[0]];
+    CPLog.debug("Cucapp is about to simulate a mouse down to the point " + locationWindowPoint);
+
+    [self _dispatchMouseEventWithType:mouseType location:locationWindowPoint modifierFlags:flags clickCount:0 window:window];
+
+    return '{"result" : "OK"}';
 }
 
-- (CPString)simulateRightClick:(CPArray)params
+- (void)simulateMouseDown:(CPArray)params
```

Comment:
`simulateMouseDown` should call internally the method `simulateMouseDownOnPoint ` once the location has been calculated